src/bin/cargo-miri.rs

```diff
@@ -441,14 +479,17 @@ fn in_cargo_miri() {
             }
             cmd.arg(arg);
         }
-        // Add `--` (to end the `cargo` flags), and then the user flags. We add markers around the
-        // user flags to be able to identify them later.  "cargo rustc" adds more stuff after this,
-        // so we have to mark both the beginning and the end.
-        cmd
-            .arg("--")
-            .arg("cargo-miri-marker-begin")
-            .args(args)
-            .arg("cargo-miri-marker-end");
+
+        let mut prefixed_args = String::new();
+        for arg in args {
+            prefixed_args += &arg.len().to_string();
+            prefixed_args.push(';');
+            prefixed_args += &arg;
+        }
```

Comment:
When inventing a serialization scheme, at the very least please put this into separate functions.

Though maybe at this point we should just use serde and serialize this as JSON, or so?